src/modules/evaluateExpression/evaluateExpression.ts

```diff
@@ -0,0 +1,128 @@
+interface IParameters {
+  [key: string]: any;
+  connection?: IConnection;
+}
+
+interface IQueryNode {
+  value?: string | number | boolean | object;
+  type?: string;
+  operator?: string;
+  children?: Array<IQueryNode>;
+}
+
+interface IConnection {
+  user?: string;
+  host?: string;
+  database?: string;
+  password?: string;
+  port?: number;
+  query?: any;
+}
+
+const defaultParameters: IParameters = {
+  connection: {},
+};
+
+export default async function evaluateExpression(
+  inputQuery: IQueryNode | string,
+  params = defaultParameters
+): Promise<string | number | boolean | any[]> {
+  // If input is JSON string, convert to Object
+  const query = typeof inputQuery === 'string' ? JSON.parse(inputQuery) : inputQuery;
+
+  // Base case
+  if (!query.children) {
+    return query.value;
+  }
+
+  // Recursive case
+  else {
+    const childrenResolved: any[] = await Promise.all(
+      query.children.map((child: IQueryNode) => evaluateExpression(child, params))
+    );
+
+    switch (query.operator) {
+      case 'AND':
+        return childrenResolved.reduce((acc: boolean, child: boolean) => {
+          return acc && child;
+        }, true);
+
+      case 'OR':
+        return childrenResolved.reduce((acc: boolean, child: boolean) => {
+          return acc || child;
+        }, false);
+
+      case 'CONCAT':
+        if (query.type === 'array') {
+          return childrenResolved.reduce((acc: any, child: any) => {
+            return acc.concat(child); // .flat(1) doesn't work for some reason
+          });
+        } else if (query.type === 'string' || !query.type) {
+          return childrenResolved.join('');
+        }
+        break;
+
+      case 'REGEX':
+        const str: string = childrenResolved[0];
+        const re: RegExp = new RegExp(childrenResolved[1]);
+        return re.test(str);
+
+      case '=':
+        return childrenResolved.every((child) => child === childrenResolved[0]);
+
+      case '!=':
+        return childrenResolved[0] !== childrenResolved[1];
+
+      case '+':
+        return childrenResolved.reduce((acc: number, child: number) => {
+          return acc + child;
+        }, 0);
+
+      case 'objectProperties':
+        try {
+          return params[childrenResolved[0].object][childrenResolved[0].property];
+        } catch {
+          return "Can't resolve object";
+        }
+
+      case 'pgSQL':
+        if (!params.connection) return 'No database connection provided';
```

Comment:
I was going to suggest to access the Pg util functions here as well. No need to pass the client here. 
I might just check how does the test work to define a mock database to use, if that can be changed if not passed as an argument to the function I guess it's alright to leave it for now

Reply:
Not sure exactly what you mean. I need to check the client object exists here or typescript gives an error on the next line saying that the connection parameter can't be `undefined`. But I have to have undefined as an option for that field as all the other operators don't require a `connection` field. So I'm just doing a type-check before calling a function. (I guess I'm also answering my own previous question though lol)